config/stack.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+	"context"
+
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/gruntwork-io/terragrunt/config/hclparse"
+	"github.com/gruntwork-io/terragrunt/internal/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+// StackConfigFile represents the structure of terragrunt.stack.hcl stack file
+type StackConfigFile struct {
+	Locals *terragruntLocal `cty:"locals"  hcl:"locals,block"`
+	Units  []*Unit          `cty:"unit" hcl:"unit,block"`
+}
+
+func ReadStackConfigFile(ctx context.Context, terragruntOptions *options.TerragruntOptions) (*StackConfigFile, error) {
+	terragruntOptions.Logger.Debugf("Reading Terragrunt stack config file at %s", terragruntOptions.TerragrungStackConfigPath)
+
+	parser := NewParsingContext(ctx, terragruntOptions)
+
+	file, err := hclparse.NewParser(parser.ParserOptions...).ParseFromFile(terragruntOptions.TerragrungStackConfigPath)
+	if err != nil {
+		return nil, errors.New(err)
+	}
+	// nolint:contextcheck
+	if err := processLocals(parser, terragruntOptions, file); err != nil {
+		return nil, errors.New(err)
+	}
+	// nolint:contextcheck
+	evalParsingContext, err := createTerragruntEvalContext(parser, file.ConfigPath)
+	if err != nil {
+		return nil, errors.New(err)
+	}
+
+	config := &StackConfigFile{}
+	if err := file.Decode(config, evalParsingContext); err != nil {
+		return nil, errors.New(err)
+	}
+
+	return config, nil
+}
+
+func processLocals(parser *ParsingContext, terragruntOptions *options.TerragruntOptions, file *hclparse.File) error {
+	localsBlock, err := file.Blocks(MetadataLocals, false)
+
+	if err != nil {
+		return errors.New(err)
+	}
+
+	if len(localsBlock) == 0 {
+		return nil
+	}
+
+	attrs, err := localsBlock[0].JustAttributes()
```

Comment:
Is this always how locals work? The first block is the only one that matters? If so, we should either throw an error if users use multiple locals blocks, or properly combine them all.

Reply:
I added a check for multiple blocks to get an error in the case of multiple locals.